src/mame/drivers/jalmah.cpp

```diff
@@ -238,11 +248,11 @@ class urashima_state : public jalmah_state
 	required_shared_ptr_array<uint16_t, 2> m_videoram;
 	required_device<gfxdecode_device> m_gfxdecode;
 	
-	uint16_t m_vreg[2][0x800];
+	std::unique_ptr<uint16_t[]> m_vreg[2];
 	tilemap_t *m_layer[2];
 	
 	TILEMAP_MAPPER_MEMBER(range0_16x16);
-	TILEMAP_MAPPER_MEMBER(range3_8x8);
+	TILEMAP_MAPPER_MEMBER(range1_8x8);
```

Comment:
range3 is an intentional naming: it follows up what was this function name before tilemap rewrite, it's not referred to the layer name itself.